### PR TITLE
add branches table

### PR DESCRIPTION
**Pull request by rkh**

### CODE REVIEW

**Comment by rkh** [comment]
I'm not 100% sure where this needs to be deployed. I guess hub? @joshk

File: db/migrate/20150223125700_create_branches.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreateBranches < ActiveRecord::Migration
+  def up
+    create_table(:branches) do |t|
+      t.integer :repository_id, null: false
+      t.integer :last_build_id
+      t.string  :name
+      t.timestamps
+    end
+    add_index(:branches, [:repository_id, :name])
```
Comment (henrikhodne) [inline db/migrate/20150223125700_create_branches.rb]: This should probably be a unique index.

Comment (rkh) [inline db/migrate/20150223125700_create_branches.rb]: Good catch. Will that error out if we happen to have two branch objects with the same name?

Comment (henrikhodne) [inline db/migrate/20150223125700_create_branches.rb]: It means that the database won't allow two rows with the same repository_id and branch name. I believe ActiveRecord raises an exception if the unique index is violated, but that depends on `save` vs `save!` if I recall correctly. Either way the save won't "succeed", so we won't have duplicates in our database.

**Comment by joshk** [comment]
This looks great to me

**Comment by joshk** [comment]
@henrikhodne how are migrations run on prod again?

**Comment by henrikhodne** [comment]
> @henrikhodne how are migrations run on prod again?

@joshk https://github.com/travis-ci/travis-migrations

**Comment by joshk** [comment]
@henrikhodne which app do we use on heroku? travis-production?

**Comment by henrikhodne** [comment]
> @henrikhodne which app do we use on heroku? travis-production?

:+1:

**Comment by joshk** [comment]
@rkh go for gold!

**Comment by rkh** [comment]
Will take care of this tomorrow.

Corresponding PR for API: https://github.com/travis-ci/travis-api/pull/164

**Pull request merged**